src/display/annotation_layer.js

```diff
@@ -500,6 +393,144 @@ var WidgetAnnotationElement = (function WidgetAnnotationElementClosure() {
 })();
 
 /**
+ * @class
+ * @alias PopupAnnotationElement
+ */
+var PopupAnnotationElement = (function PopupAnnotationElementClosure() {
+  var BACKGROUND_ENLIGHT = 0.7;
+
+  function PopupAnnotationElement(parameters) {
+    AnnotationElement.call(this, parameters);
+
+    this.pinned = false;
+  }
+
+  Util.inherit(PopupAnnotationElement, AnnotationElement, {
+    /**
+     * Render the popup annotation's HTML element in the empty container.
+     *
+     * @public
+     * @memberof PopupAnnotationElement
+     * @returns {HTMLSectionElement}
+     */
+    render: function PopupAnnotationElement_render() {
+      this.container.className = 'popupAnnotation';
+      this.container.setAttribute('hidden', true);
+
+      var content = document.createElement('div');
+      content.className = 'content';
+
+      var color = this.data.color;
+      if (color) {
+        // Enlighten the color.
+        var r = BACKGROUND_ENLIGHT * (255 - color[0]) + color[0];
+        var g = BACKGROUND_ENLIGHT * (255 - color[1]) + color[1];
+        var b = BACKGROUND_ENLIGHT * (255 - color[2]) + color[2];
+        content.style.backgroundColor = Util.makeCssRgb(r | 0, g | 0, b | 0);
+      }
+
+      var contents = this._formatContents(this.data.contents);
+      var title = document.createElement('h1');
+      title.textContent = this.data.title;
+
+      // Attach the event listeners and position the container. A parent
+      // element is only available when its type is supported.
+      var selector = '[data-annotation-id="' + this.data.parentId + '"]';
+      var parentItem = this.layer.querySelectorAll(selector)[0];
```

Comment:
Is `querySelectorAll(...)[0]` really necessary, couldn't you simply use `querySelector` instead, since those should be equivalent?